### PR TITLE
MGTransferGlobalCoarsening: add assert

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: tests/multigrid-global-coarsening/multigrid_p_01.cc [hunk 1]
```diff
@@ -129,8 +129,8 @@ test(const unsigned int n_refinements,
                                                 constraints[l]);
 
   MGTransferGlobalCoarsening<dim, VectorType> transfer(
-    transfers,
-    [&](const auto l, auto &vec) { operators[l].initialize_dof_vector(vec); });
+    transfers/*,
+    [&](const auto l, auto &vec) { operators[l].initialize_dof_vector(vec); }*/);
```
Comment (kronbichler) [inline tests/multigrid-global-coarsening/multigrid_p_01.cc:133]: Can you explain this change? Doesn't this mean that you change the code to trigger the assertion (or why isn't the call to `interpolate_to_mg` triggering it)?

Comment (peterrum) [reply tests/multigrid-global-coarsening/multigrid_p_01.cc]: > Doesn't this mean that you change the code to trigger the assertion (or why isn't the call to interpolate_to_mg triggering it)?

Yes. I wanted to test it locally. It shouldn't have got into the PR. I have removed it!

**Review by kronbichler — APPROVED** [review]

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h [hunk 2]
```diff
@@ -481,6 +481,13 @@ MGTransferGlobalCoarsening<dim, VectorType>::copy_to_mg(
 {
   (void)dof_handler;
 
+  Assert(
+    initialize_dof_vector,
+    ExcMessage(
+      "To be able to use this function, a function to initialize an internal "
+      "DoF vector has be provided in the constructor of "
```
Comment (masterleinad) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.h:488]:
```suggestion
      "DoF vector has to be provided in the constructor of "
```

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h [hunk 3]
```diff
@@ -514,6 +521,13 @@ MGTransferGlobalCoarsening<dim, VectorType>::interpolate_to_mg(
...
```
Comment (masterleinad) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.h:528]:
```suggestion
      "DoF vector has to be provided in the constructor of "
```

**Review by masterleinad — CHANGES_REQUESTED** [review]

**Comment by masterleinad** [comment]
Also, can you please rebase?

**Comment by peterrum** [comment]
@masterleinad Done! Thanks for the review!

**Review by masterleinad — APPROVED** [review]

**Pull request merged**